src/matcha_ml/core/core.py

```diff
@@ -263,6 +283,31 @@ def provision(
         return matcha_state_service.fetch_resources_from_state_file()
 
 
-def stack_set(stack: str) -> str:
-    """Placeholder for core matcha stack set functionality."""
-    return stack
+def stack_set(stack_name: str) -> None:
+    """A function for updating the stack type in the local matcha.config.json file.
+
+    Args:
+        stack_name (str): the name of the type of stack to be specified in the config file.
+    """
+
+    if stack_name.lower() not in StackType:
+        raise MatchaInputError(f"{stack_name} is not a valid stack type.")
+
+    stack_enum = StackType(stack_name)
+
+    stack_dict = {
+        "stack": {
+            "name": stack_enum.name
+        }
+    }
+
+    if MatchaConfigService.config_file_exists():
+        config = MatchaConfigService.read_matcha_config()
+        config_dict = config.to_dict()
+        config_dict.update(stack_dict)
+    else:
+        config_dict = stack_dict
+
+    config = MatchaConfig.from_dict(config_dict)
+    MatchaConfigService.write_matcha_config(config)
```

Comment:
Can we add a function in `MatchaConfigService` to handle some of this? E.g.
```
MatchaConfigService.update_config_with_dict(dict)
```

Reply:
I made the same suggestion on slack, waiting for feedback

Reply:
This would be a useful function as I am running into some issues where the {'stack'....} value is being overwritten during provisioning.